frontend/pages/hosts/ManageHostsPage/components/EnrollSecretModal/EnrollSecretModal.tsx

```diff
@@ -27,35 +28,81 @@ const EnrollSecretModal = ({
   selectedTeam,
   isPremiumTier,
   teams,
+  toggleSecretEditorModal,
+  toggleDeleteSecretModal,
+  setSelectedSecret,
+  globalSecrets,
 }: IEnrollSecretModal): JSX.Element => {
-  const globalSecret = useSelector(
-    (state: IRootState) => state.app.enrollSecret
-  );
-
   const renderTeam = () => {
     if (typeof selectedTeam === "string") {
       selectedTeam = parseInt(selectedTeam, 10);
     }
 
     if (selectedTeam === 0) {
-      return { name: "No team", secrets: globalSecret };
+      return { name: "No team", secrets: globalSecrets };
     }
     return teams.find((team) => team.id === selectedTeam);
   };
 
+  const addNewSecretClick = () => {
+    setSelectedSecret(undefined);
+    toggleSecretEditorModal();
+  };
+
   return (
-    <Modal onExit={onReturnToApp} title={"Enroll secret"} className={baseClass}>
+    <Modal
+      onExit={onReturnToApp}
+      title={"Manage enroll secrets"}
+      className={baseClass}
+    >
       <div className={baseClass}>
-        <div className={`${baseClass}__description`}>
-          Use these secret(s) to enroll devices to <b>{renderTeam()?.name}</b>:
-        </div>
-        <div className={`${baseClass}__secret-wrapper`}>
-          {isPremiumTier && (
-            <EnrollSecretTable secrets={renderTeam()?.secrets} />
-          )}
-          {!isPremiumTier && (
-            <EnrollSecretTable secrets={renderTeam()?.secrets} />
-          )}
+        {renderTeam()?.secrets?.length ? (
+          <>
+            <div className={`${baseClass}__description`}>
+              Use these secret(s) to enroll hosts to <b>{renderTeam()?.name}</b>
+              :
+            </div>
+            <div className={`${baseClass}__secret-wrapper`}>
+              {isPremiumTier && (
+                <EnrollSecretTable
+                  secrets={renderTeam()?.secrets}
+                  toggleSecretEditorModal={toggleSecretEditorModal}
+                  toggleDeleteSecretModal={toggleDeleteSecretModal}
+                  setSelectedSecret={setSelectedSecret}
+                />
+              )}
+              {!isPremiumTier && (
+                <EnrollSecretTable
+                  secrets={renderTeam()?.secrets}
+                  toggleSecretEditorModal={toggleSecretEditorModal}
+                  toggleDeleteSecretModal={toggleDeleteSecretModal}
+                  setSelectedSecret={setSelectedSecret}
+                />
+              )}
```

Comment:
It seems like this is passing the same props to `EnrollSecretTable` for both conditions of `isPremiumTier`. Do we need the `isPremiumTier` check?

Reply:
Good call!